src/Features/CSharp/Portable/Completion/CompletionProviders/SymbolCompletionProvider.cs

```diff
@@ -36,12 +37,55 @@ protected override Task<ImmutableArray<ISymbol>> GetSymbolsAsync(SyntaxContext c
                 context.SemanticModel, position, context.Workspace, options, cancellationToken);
         }
 
+        protected async override Task<bool> ShouldProvidePreselectedItemsAsync(CompletionContext completionContext, SyntaxContext syntaxContext, Document document, int position, Lazy<ImmutableArray<ITypeSymbol>> inferredTypes, OptionSet options)
+        {
+            var sourceText = await document.GetTextAsync(CancellationToken.None).ConfigureAwait(false);
+            if (ShouldTriggerInArgumentLists(sourceText, options))
+            {
+                // Avoid preselection & hard selection when triggered via insertion in an argument list.
+                if (completionContext.Trigger.Kind == CompletionTriggerKind.Insertion &&
+                    position > 0 &&
+                    await IsTriggerInArgumentListAsync(document, position - 1, CancellationToken.None).ConfigureAwait(false) == true)
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
         protected override bool IsInstrinsic(ISymbol s)
             => s is ITypeSymbol ts && ts.IsIntrinsicType();
 
         internal override bool IsInsertionTrigger(SourceText text, int characterPosition, OptionSet options)
         {
-            return CompletionUtilities.IsTriggerCharacter(text, characterPosition, options);
+            return ShouldTriggerInArgumentLists(text, options)
+                ? CompletionUtilities.IsTriggerCharacterOrArgumentListCharacter(text, characterPosition, options)
+                : CompletionUtilities.IsTriggerCharacter(text, characterPosition, options);
+        }
+
+        internal async override Task<bool> IsSyntacticTriggerCharacterAsync(Document document, int caretPosition, CompletionTrigger trigger, OptionSet options, CancellationToken cancellationToken)
+        {
+            if (trigger.Kind == CompletionTriggerKind.Insertion && caretPosition > 0)
+            {
+                var result = await IsTriggerOnDotAsync(document, caretPosition - 1, cancellationToken).ConfigureAwait(false);
+                if (result.HasValue)
+                {
+                    return result.Value;
+                }
+
+                if (ShouldTriggerInArgumentLists(document.Project.Solution.Workspace, await document.GetOptionsAsync(cancellationToken).ConfigureAwait(false)))
+                {
+                    result = await IsTriggerInArgumentListAsync(document, caretPosition - 1, cancellationToken).ConfigureAwait(false);
+                    if (result.HasValue)
+                    {
+                        return result.Value;
+                    }
+                }
+            }
```

Comment:
shouldn't there be a return false above this?  if we have an insertion trigger, and it's neither of the above cases... why do we want to still show?

Reply:
Because the trigger could just be a letter somewhere unrelated to those 2 cases.